runtime/autoload/tutor.vim

```diff
@@ -15,38 +15,28 @@ function! tutor#SetupVim()
     endif
 endfunction
 
-" Mappings: {{{1
-
-function! s:CheckMaps()
-    nmap
+" Loads metadata file, if available
+function! tutor#LoadMetadata()
+    try
+        let b:tutor_metadata = json_decode(join(readfile(expand('%').'.json'), "\n"))
+    catch
```

Comment:
Failing to load the metadata is not a failure condition, because expect regions (which atm are the only elements defined in the `.json` file, although I considered allowing defining buffer-specific mappings) are optional to the format. At worst, an error loading this will mean that the check marks and highlighting of those regions will not be performed, which is a mere visual glitch. However, as you mention, this will be mainly covered by the `filereadable()` guard, so the only kind of issue one could expect is some error in decoding the json file itself. I think I put the `try` guard in place because the errors that `json_decode` throws are often unhelpful, but I suppose it's better to show them (or a message) anyway in case something goes wrong.

I could remove this guard or do something like this:

~~~
let rjson = join(readfile(expand('%').'.json'), "\n")
try
    let b:tutor_metadata = json_decode(rjson)
catch
    echom "tutor: error loading medatada file for tutorial: ". fnamemodify(expand('%'), ':t:r')
endtry
~~~